elastic/beats
# Move service config under stackdriver.metrics and simplify metric_types

## What does this PR do?

This PR is to move service config parameter from under `stackdriver` to under `stackdriver.metrics`. Service name used to only be used for checking metadata. Now it's also been used to construct the full `metric_type` string to make google cloud API call `list.timeseries`. 

For example, if `service=compute` and metric_type is given in config as `instance/cpu/usage_time`, then the full metric.type will be `compute/googleapi.com/instance/cpu/usage_time`.

This PR also removed the `stackdriver` config layer. Now instead of `stackdriver.metrics`, configs are directly under `metrics`.

Before:
```
- module: googlecloud
  metricsets:
    - stackdriver
  project_id: "your project id"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 1m
  stackdriver:
    service: compute
    metrics:
      - aligner: ALIGN_NONE
        metric_types:
          - "compute.googleapis.com/instance/cpu/reserved_cores"
          - "compute.googleapis.com/instance/cpu/usage_time"
          - "compute.googleapis.com/instance/cpu/utilization"
          - "compute.googleapis.com/instance/uptime"
```

After:
```
- module: googlecloud
  metricsets:
    - stackdriver
  project_id: "your project id"
  credentials_file_path: "your JSON credentials file path"
  exclude_labels: false
  period: 1m
  metrics:
    - aligner: ALIGN_NONE
      service: compute
      metric_types:
        - "instance/cpu/reserved_cores"
        - "instance/cpu/usage_time"
        - "instance/cpu/utilization"
        - "instance/uptime"
```

## Checklist


## How to test this PR locally
1. Enable googlecloud metricset: `./metricbeat modules enable googlecloud`
2. Change `modules.d/googlecloud.yml` to below:
```
- module: googlecloud
  metricsets:
    - stackdriver
  project_id: elastic-observability
  credentials_file_path: "/Users/kaiyansheng/Downloads/elastic-observability-d17781618202.json"
  exclude_labels: false
  period: 1m
  metrics:
    - aligner: ALIGN_NONE
      service: compute
      metric_types:
        - "instance/cpu/reserved_cores"
        - "instance/cpu/usage_time"
        - "instance/cpu/utilization"
        - "instance/uptime"
```
3. Metrics `googlecloud.stackdriver.instance.cpu.reserved_cores.value`, `googlecloud.stackdriver.instance.cpu.usage_time.value`, `googlecloud.stackdriver.instance.cpu.utilization.value` and `googlecloud.stackdriver.instance.uptime.value` should be collected.
